.github/workflows/linux_packaging.yml

```diff
@@ -0,0 +1,368 @@
+          restore-keys: extra-${{ matrix.target }}-
+      - name: Set up Ruby
+        run: |
+          sudo apt update
+          sudo apt install -y \
+            rake \
+            ruby \
+            ruby-dev
+      - name: Get Arrow Version
+        id: get-arrow-version
+        run: |
+          # In case of dev (no workflow_dispatch inputs) compute ARROW_VERSION and
+          # use the same version for NO_RC_VERSION.
+          if [ -z "${ARROW_VERSION}" ]; then
+            ARROW_VERSION=$(git describe --tags --abbrev=0 --match "apache-arrow-[0-9]*.*" | sed 's/^apache-arrow-//')
+            ARROW_VERSION="${ARROW_VERSION}$(date +%Y%m%d)"
+            echo "ARROW_VERSION=${ARROW_VERSION}" >> $GITHUB_ENV
+            echo "NO_RC_VERSION=${ARROW_VERSION}" >> $GITHUB_ENV
+          fi
+      - name: Build
+        run: |
+          set -e
+          pushd dev/tasks/linux-packages
+          rake version:update ARROW_RELEASE_TIME="$(date --iso-8601=seconds)"
+          rake docker:pull || :
+          rake --trace ${{ matrix.task-namespace }}:build BUILD_DIR=build
+          popd
+        env:
+          APT_TARGETS: ${{ matrix.target }}
+          # TODO: Investigate what is this REPO variable and where is coming from
+          REPO: ${{ secrets.REPO }}
+          YUM_TARGETS: ${{ matrix.target }}
+      - name: Login to Dockerhub
+        if: >-
+          success() &&
+          github.event_name == 'push' &&
+          github.repository == 'apache/arrow' &&
+          github.ref_name == 'main'
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Docker Push
+        continue-on-error: true
+        if: >-
+          success() &&
+          github.event_name == 'push' &&
+          github.repository == 'apache/arrow' &&
+          github.ref_name == 'main'
+        shell: bash
+        run: |
+          pushd dev/tasks/linux-packages
+          rake docker:push
+          popd
+        env:
+          APT_TARGETS: ${{ matrix.target }}
+          REPO: ${{ secrets.REPO }}
+          YUM_TARGETS: ${{ matrix.target }}
+      - name: Build artifact paths
+        id: artifact-paths
+        shell: bash
+        run: |
+          paths=""
+          for ext in ${{ join(matrix.upload-extensions, ' ') }}; do
+            paths="$paths dev/tasks/linux-packages/*/*/repositories/**/*.${ext}"
+          done
+          echo $paths
+          echo "paths=$paths" >> $GITHUB_OUTPUT
+      - name: Upload the artifacts to the job
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+        with:
+          name: ${{ matrix.target }}-${{ matrix.architecture }}
+          path: ${{ steps.artifact-paths.outputs.paths }}
+      - name: Download Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: artifacts-downloaded
+      - name: Expected uploaded yum artifacts
```

Comment:
Should we move this and the next step to individual shell scripts (or python / ruby tests)?
This is only for yum and we will have to do something similar for `.deb` which makes the workflow slightly long to work with.

Reply:
we could also split into `linux_packaging_yum.yml` and `linux_packaging_apt.yml` and have two labels:
`CI Extra: yum linux-packaging`, `CI Extra: apt linux-packaging` to reduce the amount of jobs